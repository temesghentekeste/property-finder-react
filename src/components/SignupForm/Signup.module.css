:root {
    --primary-color: #ec5323;
}

.content {
  position: relative; 
  z-index: 2;
}

.formContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
  text-align: center;
}


.formContainer__formInput {
  margin-bottom: 1.5rem;
}

.formContainer__formInput input {
  border: 1px solid var(--primary-color);
  border-radius: 10px;
  font-size: 20px;
  padding: 4px 7px;
  outline: 0;
  background: transparent;
}

.formContainer button {
  padding: 0.4rem 1.4rem;
  background: var(--primary-color);
  border: none;
  border-radius: 10px;
  font-weight: bold;
  color: #fff;
  font-size: 0.8rem;

}

.formContainer_header {
  margin-bottom: 3.4rem;
  text-align: center;
  width: 70%;
  color: #444;
}

.formContainer_header h2 {
  font-size: 1.8rem;
  font-weight: 600;
}

.formContainer_header p {
  font-style: 0.9rem;
  font-weight: normal;
}
