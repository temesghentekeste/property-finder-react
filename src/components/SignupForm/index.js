import { useState, useEffect, useRef } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import HomeIcon from '@material-ui/icons/Home';
import styles from './Signup.module.css';
import common from '../../common/common.module.css';

const SignupForm = ({ handleSubmit }) => {
  const [user, setUser] = useState({
    username: '',
    password: '',
  });

  const usernameRef = useRef();
  const handleChange = (e) => {
    const { name, value } = e.target;

    setUser({ ...user, [name]: value });
  };

  useEffect(() => {
    usernameRef.current.focus();
  }, []);

  return (
    <div className={`${styles.formContainer} ${styles.content}`}>
      <header className={styles.formContainer_header}>

        <h2>Sign Up</h2>
        <p>Hello there! Sign up and enjoy wonderful properies.</p>
      </header>
      <form onSubmit={(e) => handleSubmit(e, user)}>
        <div className={styles.formContainer__formInput}>
          <input
            type="text"
            id="username"
            name="username"
            value={user.username}
            onChange={handleChange}
            ref={usernameRef}
          />
        </div>
        <div className={styles.formContainer__formInput}>
          <input
            type="password"
            id="password"
            name="password"
            value={user.password}
            onChange={handleChange}
          />
        </div>
        <div className={styles.formContainer__formInput}>
          <button type="submit" className={styles.formContainer_btn}>
            Sign Up
          </button>
        </div>
      </form>
      <div className={styles.signin}>
        <span>If you already have an account,&nbsp;</span>
        <Link to="/login">sign in</Link>
      </div>
      <div className={common.backToHome}>
        <Link to="/">
          Back to&nbsp;
          <HomeIcon />
        </Link>
      </div>
    </div>
  );
};

SignupForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,

};

export default SignupForm;
