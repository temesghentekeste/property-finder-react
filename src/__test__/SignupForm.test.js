/* eslint-disable no-console */
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import SignupForm from '../components/SignupForm';

let getByTestId;
const handleSubmit = () => console.log('Handled');
beforeEach(() => {
  const component = render(
    <Router>
      <SignupForm handleSubmit={handleSubmit} />
    </Router>,
  );
  getByTestId = component.getByTestId;
});

it('renders the LoginForm component', () => {
  const component = renderer
    .create(
      <Router>
        <SignupForm handleSubmit={handleSubmit} />
      </Router>,
    )
    .toJSON();

  expect(component).toMatchSnapshot();
});

test('should render SignupForm component with correct heading', () => {
  const headingText = getByTestId('SignupForm-heading');

  expect(headingText).toHaveTextContent('Sign Up');
});
