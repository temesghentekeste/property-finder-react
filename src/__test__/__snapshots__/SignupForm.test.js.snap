// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the LoginForm component 1`] = `
<div
  className="formContainer content"
  data-testid="SignupForm"
>
  <header
    className="formContainer_header"
  >
    <h2
      data-testid="SignupForm-heading"
    >
      Sign Up
    </h2>
    <p>
      Hello there! Sign up and enjoy wonderful properies.
    </p>
  </header>
  <form
    onSubmit={[Function]}
  >
    <div
      className="formContainer__formInput"
    >
      <input
        id="username"
        name="username"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="formContainer__formInput"
    >
      <input
        id="password"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="formContainer__formInput"
    >
      <button
        className="formContainer_btn"
        type="submit"
      >
        Sign Up
      </button>
    </div>
  </form>
  <div
    className="signin"
  >
    <span>
      If you already have an account, 
    </span>
    <a
      href="/login"
      onClick={[Function]}
    >
      sign in
    </a>
  </div>
  <div
    className="backToHome"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Back to 
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
    </a>
  </div>
</div>
`;
